File: AlphaStandardDemo/Interfaces/IFabrica.vb
Imports AlphaStandardDemo.Models

Namespace Interfaces
  Public Interface IFabrica
    Property Carro As Carro
    Property Entregues As Integer
    Property ListaCarrosEntregues As List(Of String)
    Sub CriarCarro()
    Sub PintarCarro(color As Drawing.Color)
    Sub MudarMarca(marca As String)
    Sub MudarModelo(modelo As String)
    Sub MudarAno(ano As Integer)
    Sub EntregarCarro()
  End Interface
End Namespace

File: AlphaLib/Interfaces/IFabrica.cs
using AlphaLib.Models;
using System.Collections.Generic;

namespace AlphaLib.Interfaces
{
    public interface IFabrica
    {
        Carro Carro { get; set; }
        int Entregues { get; set; }
        List<string> ListaCarrosEntregues { get; set; }
        void CriarCarro();
        void PintarCarro(string color);
        void MudarMarca(string marca);
        void MudarModelo(string modelo);
        void MudarAno(int ano);
        void EntregarCarro();
    }
}
